crates/holochain_p2p/tests/tests/blocks.rs

```diff
@@ -130,199 +109,246 @@ async fn agent_is_removed_from_peer_store_when_blocked() {
     assert!(peer_store.get(agent.to_k2_agent()).await.unwrap().is_none());
 }
 
-#[tokio::test(flavor = "multi_thread")]
-async fn are_all_blocked_mixed_then_all_blocked() {
-    let dna_hash = fixt!(DnaHash);
-    let agent_1 = fixt!(AgentPubKey);
-    let agent_2 = fixt!(AgentPubKey);
-    let cell_id_1 = CellId::new(dna_hash.clone(), agent_1.clone());
-    let cell_id_2 = CellId::new(dna_hash.clone(), agent_2.clone());
-
-    let TestCase {
-        actor,
-        blocks_module,
-    } = TestCase::new(&dna_hash).await;
-
-    // Initially not blocked.
-    assert!(!blocks_module
-        .are_all_blocked(vec![
-            kitsune2_api::BlockTarget::Agent(agent_1.to_k2_agent()),
-            kitsune2_api::BlockTarget::Agent(agent_2.to_k2_agent()),
-        ])
-        .await
-        .unwrap());
+mod blocks_impl {
+    use super::*;
 
-    // Block agent1 only.
-    actor
-        .block(Block::new(
-            BlockTarget::Cell(cell_id_1, CellBlockReason::BadCrypto),
-            InclusiveTimestampInterval::try_new(Timestamp::now(), Timestamp::max()).unwrap(),
-        ))
-        .await
-        .unwrap();
+    #[tokio::test(flavor = "multi_thread")]
+    async fn block_someone() {
+        let dna_hash = fixt!(DnaHash);
+        let agent = fixt!(AgentPubKey);
+        let cell_id = CellId::new(dna_hash.clone(), agent.clone());
 
-    // Mixed list should yield false.
-    assert!(!blocks_module
-        .are_all_blocked(vec![
-            kitsune2_api::BlockTarget::Agent(agent_1.to_k2_agent()),
-            kitsune2_api::BlockTarget::Agent(agent_2.to_k2_agent()),
-        ])
-        .await
-        .unwrap());
+        let TestCase {
+            actor,
+            blocks_module,
+        } = TestCase::new(&dna_hash).await;
 
-    // Block agent2 as well.
-    actor
-        .block(Block::new(
-            BlockTarget::Cell(cell_id_2, CellBlockReason::BadCrypto),
-            InclusiveTimestampInterval::try_new(Timestamp::now(), Timestamp::max()).unwrap(),
-        ))
-        .await
-        .unwrap();
+        assert!(!blocks_module
+            .is_blocked(kitsune2_api::BlockTarget::Agent(agent.to_k2_agent()))
+            .await
+            .unwrap());
+        assert!(!blocks_module.are_all_blocked(vec![]).await.unwrap());
```

Comment:
I don't follow what this is trying to assert - to me it seems like are_all_blocked should error due to invalid input if you pass it an empty list.

Reply:
Yes, we've had this conversation when the tests were written and in kitsune where this is exclusively called, the function errors when an empty vec is passed in. So this case currently can't happen.